storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -252,26 +257,43 @@ public SafeFuture<Optional<BeaconState>> getStateAtSlotExact(final UInt64 slot)
   }
 
   public SafeFuture<Optional<BeaconState>> getStateForBlockProduction(
```

Comment:
I think it is worth adding a doc comment that mentions how this method can potentially start a state regeneration to handle the LBR.